### PR TITLE
3.0: move boto3 calls from cluster validator to boto3 layer

### PR DESCRIPTION
**Pull request by hanwen-pcluste**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: cli/src/pcluster/validators/awsbatch_validators.py [hunk 1]
```diff
@@ -122,8 +113,7 @@ def _get_supported_batch_instance_types():
     known_exceptions = ["optimal"]
     supported_instance_types_and_families = supported_instance_types + supported_instance_families + known_exceptions
     try:
-        emsg = _get_cce_emsg_containing_supported_instance_types()
-        parsed_instance_types_and_families = _parse_supported_instance_types_and_families_from_cce_emsg(emsg)
+        parsed_instance_types_and_families = AWSApi.instance().batch.get_parsed_supported_instance_types_and_families()
```
[the remark this answers is not in the document]
Comment (hanwen-pcluste) [reply cli/src/pcluster/validators/awsbatch_validators.py]: Done

File: cli/src/pcluster/templates/imagebuilder_stack.py [hunk 2]
```diff
@@ -30,6 +30,8 @@
 from aws_cdk import aws_sns as sns
 from aws_cdk.core import CfnParameter, CfnTag, Construct, Fn, Stack
 
+import pcluster.aws.aws_api
+import pcluster.aws.common
```
[the remark this answers is not in the document]
Comment (hanwen-pcluste) [reply cli/src/pcluster/templates/imagebuilder_stack.py]: Done

File: cli/src/pcluster/cli_commands/update.py [hunk 3]
```diff
@@ -15,6 +15,8 @@
 
 from tabulate import tabulate
 
+import pcluster.aws.aws_api
+import pcluster.aws.common
```
[the remark this answers is not in the document]
Comment (hanwen-pcluste) [reply cli/src/pcluster/cli_commands/update.py]: Done

File: cli/src/pcluster/cli_commands/commands.py [hunk 4]
```diff
@@ -22,6 +22,8 @@
 from botocore.exceptions import ClientError
 from tabulate import tabulate
 
+import pcluster.aws.aws_api
+import pcluster.aws.common
```
[the remark this answers is not in the document]
Comment (hanwen-pcluste) [reply cli/src/pcluster/cli_commands/commands.py]: Done

File: cli/src/pcluster/aws/ec2.py [hunk 5]
```diff
@@ -335,3 +335,25 @@ def get_ebs_snapshot_info(self, ebs_snapshot_id):
         }
         """
         return self._client.describe_snapshots(SnapshotIds=[ebs_snapshot_id]).get("Snapshots")[0]
+
+    @AWSExceptionHandler.handle_client_exception
+    def describe_single_security_group(self, security_group_id):
```
[the remark this answers is not in the document]
Comment (hanwen-pcluste) [reply cli/src/pcluster/aws/ec2.py]: Done

**Review by enrico-usai — APPROVED** [review]

**Pull request merged**